src/api/forms/service.js

```diff
@@ -163,12 +164,36 @@ export async function createForm(metadataInput, author) {
 }
 
 /**
- * Lists the available form metadata
+ * Lists forms and returns query result metadata (e.g., pagination details, total counts)
+ * @param {PaginationOptions} [options] - Optional pagination options
+ * @returns {Promise<FormMetadata[] | QueryResult<FormMetadata>>}
```

Comment:
Unfortunately I had to remove the `@overload` due to Jest not picking up the second promise (shifing orders doesn't help either 😭 ) and the linter failing in the tests in `form.test.js` which check for the second response (an array of `FormMetadata`) when pagination options aren't used. A bit of a tradeoff, but I consequently had to type the pagination response objects in `service.test.ts` 😩